core/drand_daemon_control.go

```diff
@@ -243,22 +247,47 @@ func (dd *DrandDaemon) Stop(ctx context.Context) {
 		dd.log.Infow("Stopping DrandDaemon")
 	}
 	for _, bp := range dd.beaconProcesses {
-		dd.log.Debugw("Sending Stop to beaconProcesses", "bp", bp.beaconID)
+		dd.log.Debugw("Sending Stop to beaconProcesses", "id", bp.getBeaconID())
 		bp.Stop(ctx)
 	}
 
+	// TODO (dlsniper): Should we wait here for all BPs to finish or run ahead and close the daemon?
```

Comment:
we could allow a grace period after which we force kill

Reply:
okay nvm we do that already ^^

Reply:
Yes, this was more a flag for reviewers kind of todo. It seems that having a timeout followed by a force shutdown is desirable. Correct?

Reply:
Yes 👍🏻